code/modules/fishing/fish_catalog.dm

```diff
@@ -0,0 +1,110 @@
+///Book detailing where to get the fish and their properties.
+/obj/item/book/fish_catalog
+	name = "Fish Encyclopedia"
+	desc = "Indexes all fish known to mankind (and related species)."
+	icon_state = "fishbook"
+	starting_content = "Lot of fish stuff" //book wrappers could use cleaning so this is not necessary
+
+/obj/item/book/fish_catalog/on_read(mob/user)
+	ui_interact(user)
+
+/obj/item/book/fish_catalog/ui_interact(mob/user, datum/tgui/ui)
+	. = ..()
```

Comment:
This may give people some bad ideas, so I'd just remove it if I were you because you shouldn't really be calling the parent in this proc (it could end up opening multiple UIs)